extensions/wikia/ImageReview/ImageReviewSpecialController.class.php

```diff
@@ -280,4 +203,112 @@ protected function parseImageData($data) {
 
 		return $images;
 	}
+
+	private function handlePostAction() {
+		$data = $this->wg->request->getValues();
+		if ( !empty($data) ) {
+			$images = $this->parseImageData($data);
+
+			if ( count($images) > 0 ) {
+				$this->helper->updateImageState( $images, $this->action );
+			}
+			$this->ts = null;
+		}
+	}
+
+	private function getOrderingMethod() {
+		if ($this->wg->user->isAllowed('imagereviewcontrols')) {
+			$order = (int)$this->getVal('sort', -1);
+			if ($order >= 0) {
+				$this->app->wg->User->setGlobalPreference('imageReviewSort', $order);
+				$this->app->wg->User->saveSettings();
+			}
+
+			$order = $this->app->wg->User->getGlobalPreference('imageReviewSort');
+			return $order;
+		} else {
+			$order = -1;
+			return $order;
+		}
+	}
+
+	private function parseAction() {
+		$actions = explode( '/', $this->getPar() );
+		return array_pop( $actions );
+	}
+
+	private function checkUserPermissions() {
```

Comment:
this method doesn't return value when use has permission.
Please add return true in the end.